pkg/agent/openflow/client.go

```diff
@@ -313,6 +313,41 @@ func (c *client) addFlows(cache *flowCategoryCache, flowCacheKey string, flows [
 	return nil
 }
 
+// addFlowsWithMultipleKeys installs the flows with different flowCache keys and adds them into the cache on success.
+// Same as addFlows, it will skip flows whose cache already exists. All flows will be installed via a bundle.
+func (c *client) addFlowsWithMultipleKeys(cache *flowCategoryCache, keyToFlows map[string][]binding.Flow) error {
```

Comment:
it kinds of feels like this function should either call `addFlows` and re-use the code, or should replace it altogether. Am I missing something?

Reply:
You are right, `addFlows` can leverage `addFlowsWithMultipleKeys`. I have made the change to reduce code redundancy.